hint_js.go

```diff
@@ -7,8 +7,14 @@ var hints = make(map[Hint]int)
 type Hint int
 
 const (
-	AlphaBits Hint = 0x00021004
-	Samples   Hint = 0x0002100D
+	AlphaBits                       Hint = 0x00021004
+	Samples                         Hint = 0x0002100D
+	Depth                           Hint = 0x00021100 // github.com/goxjs/glfw original hint for JS
+	Stencil                         Hint = 0x00021101 // github.com/goxjs/glfw original hint for JS
+	PremultipliedAlpha              Hint = 0x00021102 // github.com/goxjs/glfw original hint for JS
+	PreserveDrawingBuffer           Hint = 0x00021103 // github.com/goxjs/glfw original hint for JS
+	PreferLowPowerToHighPerformance Hint = 0x00021104 // github.com/goxjs/glfw original hint for JS
+	FailIfMajorPerformanceCaveat    Hint = 0x00021105 // github.com/goxjs/glfw original hint for JS
```

Comment:
Hmm. I'm a little concerned that apps that use these will only be able to compile for js architecture and not desktop. Maybe we should add these to `hint_glfw.go` and have them be ignored (until they can be accepted by the desktop implementations)? I think hints by definition are optional, so if an implementation doesn't support the given hint, ignoring it (rather than producing a fatal error) seems acceptable.